src/Articulate/UmbracoHelperExtensions.cs

```diff
@@ -264,12 +250,77 @@ internal static PostsByTagModel GetContentByTag(this UmbracoHelper helper, IMast
             return getResult();
 #else
             //cache this result for a short amount of time
-            return appContext.ApplicationCache.RuntimeCache.GetCacheItem<PostsByTagModel>(
+            
+            return (PostsByTagModel) appContext.ApplicationCache.RuntimeCache.GetCacheItem(
                 string.Concat(typeof(UmbracoHelperExtensions).Name, "GetContentByTag", masterModel.RootBlogNode.Id, tagGroup),
                 getResult, TimeSpan.FromSeconds(30));
 #endif
         }
 
+        internal static IEnumerable<PostModel> GetContentByAuthor(this UmbracoHelper helper, AuthorModel author)
+        {
+            var listNode = GetListNode(author);
+
+            var posts = listNode.Children
+                .Where(
+                    c =>
+                        string.Equals(c.GetPropertyValue<string>("author"), author.Name.Replace("-", " "),
+                            StringComparison.InvariantCultureIgnoreCase))
+                .Select(c => new PostModel(c))
+                .OrderByDescending(c => c.PublishedDate)
+                .ToList();
+
+            return posts;
+        }
+
+        private static IPublishedContent GetListNode(IMasterModel masterModel)
+        {
+            var listNode = masterModel.RootBlogNode.Children
+                .FirstOrDefault(x => x.DocumentTypeAlias.InvariantEquals("ArticulateArchive"));
+            if (listNode == null)
+            {
+                throw new InvalidOperationException(
+                    "An ArticulateArchive document must exist under the root Articulate document");
+            }
+
+            return listNode;
+        }
+
+        internal static IEnumerable<AuthorModel> GetContentByAuthors(this UmbracoHelper helper, AuthorListModel authorList)
```

Comment:
Unfortunately this is going to be horribly inefficient if there are thousands of blog posts. This would be much better done with an XPathNodeIterator: https://our.umbraco.org/documentation/reference/Common-Pitfalls/#too-much-linq-xpath-is-still-your-friend